library/core/src/main/java/com/google/android/exoplayer2/upstream/cache/SimpleCache.java

```diff
@@ -408,18 +408,15 @@ public synchronized File startFile(String key, long position, long length) throw
     CachedContent cachedContent = contentIndex.get(key);
     Assertions.checkNotNull(cachedContent);
     Assertions.checkState(cachedContent.isFullyLocked(position, length));
-    if (!cacheDir.exists()) {
-      // The cache directory has been deleted from underneath us. Recreate it, and remove in-memory
+    if (createCacheDirectories(cacheDir)) {
```

Comment:
Checking the directory is really a directory every time a file is started seems like overkill. What is the problem this is actually trying to solve? The user replacing the cache directory with a file, whilst the app is simultaneously running, seems like something that would never happen in practice.

I think I preferred the original, more minimal version of this change. Could we consider restoring the pull request to that version?

I see now my suggestion to check `isDirectory` may not have been a good suggestion, since it kind of implied that it would make sense to do this. But on reflection I'm not sure what it's really helping with. Apologies for that!